### PR TITLE
Use GtkSource.view for taskview and enable undo

### LINKED ISSUE #296: Implement undo & redo in the "task editor" window
In most (all?) text editors hitting ctrl+z undo the last action, like word or character typed/removed, etc...

This has never worked in GTG, old or current master version. So for example removing a task's content by mistake needs to be re-typed.

### PR DESCRIPTION
**Pull request by diegogangl**
This PR makes taskview sublcass GtkSource.view instead of the regular textview.
The immediate change is that now we can have undo/redo. It also opens the door for more features, like tag autocompletion.
It's a small enough change that can go into 0.6

The only limitation for now is that you can't undo a subtask once it's created. I don't think this is particularly bad though, since ctrl+z'ing a subtask would delete it without confirmation.

closes #296

### CODE REVIEW

**Comment by Neui** [comment]
Don't forget to modify the README dependency list

File: README.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ git pull --rebase
   - pango
   - gdk-pixbuf
   - GTK 3
+  - GtkSource 4
```
Comment (Neui) [inline README.md:53]: The project name is GtkSourceView 4, might better to use that name

File: GTG/gtk/editor/taskview.py [hunk 2]
```diff
@@ -188,6 +188,7 @@ def process(self) -> None:
 
         # Parse the text line by line until the end of the buffer
         while not start.is_end():
+
```
Comment (Neui) [inline GTG/gtk/editor/taskview.py:191]: Unnecessary empty line.

**Review by Neui — APPROVED** [review]
Works.

**Comment by diegogangl** [comment]
Thanks for reviewing man!

**Pull request merged**